lib/ruby-enum/enum.rb

```diff
@@ -43,8 +45,10 @@ def validate_value!(value)
       end
 
       def const_missing(key)
-        if @_enum_hash[key]
+        if @_enum_hash && @_enum_hash[key]
```

Comment:
Maybe `.key?(key)`? I believe it's faster. Otherwise assign first to avoid the double lookup:

```ruby
r = @_enum_hash[key] if @_enum_hash
if r
  r.value
elsif ...
```